profile/src/main/java/com/scottlogic/deg/profile/reader/names/NameRetrievalService.java

```diff
@@ -35,12 +35,35 @@
 
     @Override
     public Set<NameFrequencyHolder> retrieveValues(NameConstraintTypes configuration) {
-        return NAME_TYPE_MAPPINGS.get(configuration).stream()
+        switch (configuration) {
+            case FIRST:
```

Comment:
Is there meant to be a return in this case?

Reply:
No.
First or last use one routing for logic (generate singles) which is resolved from getting their configuration from the map.
Full generates the combination of the two.

Reply:
Ok sure. You could possibly add a comment explaining that, but up to you.

Reply:
Looking at https://google.github.io/styleguide/javaguide.html#s4.8.4-switch (which isn't strictly our style guide but a good metric), it states that it's a good idea to mark fall through if logic occurs, but unnecessary for each statement group. I'm inclined to agree with this, but very open to arguments otherwise.